litestar/channels/redis.py

```diff
@@ -0,0 +1,220 @@
+            This operation is performed atomically, using a lua script
+        """
+        await self._publish_script(keys=list(set(channels)), args=[data])
+
+    async def stream_events(self) -> AsyncGenerator[tuple[str, Any], None]:
+        """Return a generator, iterating over events of subscribed channels as they become available.
+
+        If no channels have been subscribed to yet via :meth:`subscribe`, sleep for ``stream_sleep_no_subscriptions``
+        milliseconds.
+        """
+
+        while True:
+            if not self._pub_sub.subscribed:
+                await asyncio.sleep(self._stream_sleep_no_subscriptions)  # no subscriptions found so we sleep a bit
+                continue
+            message = await self._pub_sub.get_message(ignore_subscribe_messages=True, timeout=None)  # type: ignore[arg-type]
+            if message is None:
+                continue
+            channel = message["channel"].decode()
+            data = message["data"]
+            yield channel, data
+
+    async def get_history(self, channel: str, limit: int | None = None) -> list[bytes]:
+        """Not implemented"""
+        raise NotImplementedError()
+
+
+class RedisChannelsStreamBackend(RedisChannelsBackend):
+    def __init__(
+        self,
+        history: int,
+        *,
+        redis: Redis,
+        stream_sleep_no_subscriptions: int = 1,
+        cap_streams_approximate: bool = True,
+        stream_ttl: int | timedelta = timedelta(seconds=60),
+        key_prefix: str = "LITESTAR_CHANNELS",
+    ) -> None:
+        """Redis channels backend, `streams https://redis.io/docs/data-types/streams/>`_.
+
+        Args:
+            history: Amount of messages to keep. This will set a ``MAXLEN`` to the streams
+            redis: A :class:`redis.asyncio.Redis` instance
+            key_prefix: Key prefix to use for streams
+            stream_sleep_no_subscriptions: Amount of time in milliseconds to pause the
+                :meth:`stream_events` generator, should no subscribers exist
+            cap_streams_approximate: Set the streams ``MAXLEN`` using the ``~`` approximation
+                operator for improved performance
+            stream_ttl: TTL of a stream in milliseconds or as a timedelta. A streams TTL will be set on each publishing
+                operation using ``PEXPIRE``
+        """
+        super().__init__(
+            redis=redis, stream_sleep_no_subscriptions=stream_sleep_no_subscriptions, key_prefix=key_prefix
+        )
+
+        self._history_limit = history
+        self._subscribed_channels: set[str] = set()
+        self._cap_streams_approximate = cap_streams_approximate
+        self._stream_ttl = stream_ttl if isinstance(stream_ttl, int) else int(stream_ttl.total_seconds() * 1000)
+        self._flush_all_streams_script = self._redis.register_script(_FLUSHALL_STREAMS_SCRIPT)
+        self._publish_script = self._redis.register_script(_XADD_EXPIRE_SCRIPT)
+
+    async def on_startup(self) -> None:
```

Comment:
Why do you need this boilerplate?

Reply:
Because it's an abstract method on the base class, and in this implementation it's a no-op. Alternative would be to implement this as no-op in `RedisChannelsBackend` and then overwrite again here but this seems more explicit since you won't have to potentially consider `super()` calls etc. It just seems cleaner to be to only implement abstract methods once.